### ISSUE TITLE
UX: Reordering nested menus doesn’t match visually in front-end

### ISSUE THREAD

**Issue by Dinamiko** [voldemortensen/menu-customizer#106]
I did a quick usability test working with nested menus, the main issue I found was that after doing some drags and drops and reordering, the result doesn’t match visually in frontend. here is a screenshot:
![menu_customizer_notmatch_frontend](https://cloud.githubusercontent.com/assets/456223/8144006/e6637336-11c8-11e5-8f5f-bfcd974ea813.jpg)

this is the video I recorded:
https://vimeo.com/130615602

I noticed some other UX stuff:

Compared to appearence / menus, the experience with reordering items it's far less user friendly, this was my first try with menu customizer plugin but doing something trivial as reordering some items took me a lot of time compared to the Appearence / menus.
this is a video I recorded doing some reordering in appearance  / menus:
https://vimeo.com/130615710

The menu customizer preview doesn't match even saving the menu. I tested this in Twenty Fifteen and Make themes, here are a screenshot:
![menu_customizer_preview](https://cloud.githubusercontent.com/assets/456223/8143998/ce39c2d8-11c8-11e5-9a9e-3927d776a181.jpg)

**Comment by westonruter** [comment]
Did you try on the feature/issue-67 branch?

**Comment by Dinamiko** [comment]
Hi westonruter, I installed the master branch and then click update plugin because of the message 'there is a new version...', so I'm testing with the Version 0.5-20150613 right now.

**Comment by westonruter** [comment]
The issue is probably fixed in the PR #73 we have opened, but it depends on some Core patches to work properly, so it can be difficult to test. You have to checkout the `feature/issue-67` branch, and not opt for the plugin upgrade via the WP Admin, and then you have to checkout the `feature/menu-customizer` branch of Core itself, as found at https://github.com/xwp/wordpress-develop/pull/91

**Comment by designsimply** [comment]
Thanks for testing, Dinamiko!

You should be able to re-test with the latest core trunk and plugin now because some core patches as well as PR #73 for the plugin were merged after your original test was done.

I updated my local WP and plugin and tried testing to see if I could reproduce the problems showing in your video (thanks for including it!), and I found that drag and drop as well as the live preview in Twenty Fifteen was much improved, but I do think there still may be a bug lurking, however, I haven't been able to pin it down to get some reliable steps to reproduce. If I open the customizer and set a previously saved menu as primary and then start moving things around quickly, I can usually eventually see this error show up in the console in Firefox 38.0.5 on Mac OSX 10.10.3:

`Uncaught TypeError: Cannot assign to read only property 'position' of false
menu-customizer.js?ver=4.3-alpha-32280-src:1789`

I'm including this screenshot for reference, but I may need some help trying to figure out exactly what triggers the error and whether it's related to the same problems showing in the original user test linked earlier in this issue.

![screen shot 2015-06-14 at sun jun 14 10 06 08 pm](https://cloud.githubusercontent.com/assets/1119271/8151817/4d3f6ce4-12e2-11e5-87b4-2687fcf64c18.png)
<sup>Seen at http://src.wordpress-develop.dev/wp-admin/customize.php tested with WordPress 4.3-alpha-32775 and Menu Customizer plugin 0.6 (cef7fbc) using Firefox 38.0.5 on Mac OSX 10.10.3.</sup>

**Comment by designsimply** [comment]
Moving this to the `Post-merge` milestone until more reliable steps to reproduce can be found.

**Comment by westonruter** [comment]
@designsimply I was able to reproduce the JS error you noted. I have a pending fix here: https://github.com/voldemortensen/menu-customizer/commit/0ce3c193d5467d99aa449628a4b97e77b6c21cf5 as part of #112.

**Comment by maddisondesigns** [comment]
I posted this same issue (and a number of others), 6 days ago over on the official forum for the plugin
https://wordpress.org/support/topic/drag-n-drop-functionality-not-working-properly?replies=1

**Comment by westonruter** [comment]
@maddisondesigns All of these issues should be fixed with the most recent plugin release, once this PR is merged: https://github.com/voldemortensen/menu-customizer/pull/112

Note that you'll need to be running Core on trunk.

**Comment by designsimply** [comment]
These issues have been fixed and have now been merged into WordPress core trunk, so the plugin is no longer needed for testing if you have the latest WordPress nightly build.
